galsim/utilities.py

```diff
@@ -589,17 +590,35 @@ def horner(x, coef, dtype=None):
 
     @returns a numpy array of the evaluated polynomial.  Will be the same shape as x.
     """
-    coef = np.trim_zeros(coef, trim='b')
-    result = np.zeros_like(x, dtype=dtype)
-    if len(coef) == 0: return result
-    result += coef[-1]
-    for c in coef[-2::-1]:
-        result *= x
-        if c != 0: result += c
-    #np.testing.assert_almost_equal(result, np.polynomial.polynomial.polyval(x,coef))
+    result = np.empty_like(x, dtype=dtype)
+    # Make sure everything is an array
+    if result.dtype == float:
```

Comment:
Do we ever not want a float?

Reply:
Not in anything we currently do.  But I can imagine wanting to do this calculation for int or complex outputs.